plugins/qcheck-stm/doc/index.mld

```diff
@@ -470,3 +470,45 @@ and peculiarities:
         Whenever a functions returns a SUT, but does not take a SUT as an input
         argument, any [Gen.int] generators in [Spec.arb_cmd] are automatically
         exchanged with [Gen.small_signed_int]}}
+
+{1 Segmentation faults and non-termination}
+
+In the standard configuration, the code generated by the [qcheck-stm] plugin
+will run a number of tests in sequence in the same process. If one of those
+test runs crashes (e.g. by incorrectly using [Obj.magic]), this automatically
+also kills the overall process, therefore no trace can be shown to the user of
+which series of commands lead to the crashing scenario. Similarily, if one of
+the functions under test does not return, the overall test suite is stalled
+indefinitely.
+
+It is possible to instruct the underlying runtime to instead run all tests in
+separate processes with a given timeout, so that in case of a segmentation fault
+the runtime can still recover a trace leading to said crash, and non-terminating
+computations are automatically stopped after the given timeout has passed.
+
+This is done by setting the environment variable [ORTAC_QCHECK_STM_TIMEOUT] to a
+timeout value in seconds, after which the runtime will automatically kill the
+process running the test.
+
+For example when using dune to run the test, this can be achieved in the
+following way:
+
+{@dune[
+(test
+ (name stm_example)
+ (libraries
+  qcheck-core
+  qcheck-core.runner
+  qcheck-stm.stm
+  qcheck-stm.sequential
+  qcheck-multicoretests-util
+  ortac-runtime
+  example)
+ (action
+  (setenv
+   ORTAC_QCHECK_STM_TIMEOUT
+   10
+    (run %{test} --verbose)))
+]}
```

Comment:
One can also set environment variables on the command line and have them be visible in dune:
```
$ ORTAC_QCHECK_STM_TIMEOUT=10 dune test ...
```
I think that offers a reasonable workflow when one hits an unexpected segfault - just rerun the last command in the console with a small addition :slightly_smiling_face:

Reply:
Indeed! Maybe we should distinguish between the CLI workflow and the CI workflow?

Reply:
When I updated the documentation I was actually considering if I should mention other ways of setting environment variables, or if the standard user would already know about those. But I will add some additional comments about that!